gpx.js

```diff
@@ -322,12 +322,13 @@ L.GPX = L.FeatureGroup.extend({
     for (j = 0; j < tags.length; j++) {
       el = xml.getElementsByTagName(tags[j][0]);
       for (i = 0; i < el.length; i++) {
-        var coords = this._parse_trkseg(el[i], xml, options, tags[j][1]);
+        var coords = this._parse_trkseg(el[i], options, tags[j][1]);
         if (coords.length === 0) continue;
 
         // add track
         var l = new L.Polyline(coords, options.polyline_options);
-        this.fire('addline', { line: l })
+        this.fire('parsesegment', { line: l, element: el[i] });
```

Comment:
Why duplicate the events? I think you can just add the relevant `element` to the `addline` and `addpoint` events.